src/app/comments/comment-form-container/comment-form-container.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute} from "@angular/router";
+import {CommentService} from "../comment.service";
+import {NewComment} from "../newComment.interface";
+import {switchMap, tap} from "rxjs/operators";
+
+
+@Component({
+  selector: 'app-comment-form-container',
+  templateUrl: './comment-form-container.component.html',
+  styleUrls: ['./comment-form-container.component.css']
+})
+export class CommentFormContainerComponent implements OnInit {
+
+  ngOnInit(): void {
+  }
+
+  constructor(private route: ActivatedRoute, private commentService: CommentService ){
+  }
+
+  errorMessage:string = '';
+  successfulSending:boolean;
+
+  comment: NewComment = {
+    body: null,
+    postId: null,
+    firstName: null,
+    lastName: null,
+    email: null,
+  }
+
+  getCommentDataFromParent(data){
+    this.comment.firstName = data.firstName;
+    this.comment.lastName = data.lastName;
+    this.comment.email = data.email;
+    this.comment.body = data.body;
+
+    this.route.params.pipe( tap(params => {
+        this.comment.postId = parseInt(params.id);
```

Comment:
this step is not necessary, because here `this.commentService.addComment(this.comment, params.id)))` after saving the comment data will get the postId anyway in the db.json. The json-server makes this assocation for you